Formula/catch2.rb

```diff
@@ -17,11 +18,9 @@ class Catch2 < Formula
 
   depends_on "cmake" => :build
 
-  fails_with gcc: "5"
-
   def install
     mkdir "build" do
-      system "cmake", "..", "-DBUILD_TESTING=OFF", *std_cmake_args
+      system "cmake", "..", "-DBUILD_TESTING=OFF", "-DCMAKE_CXX_STANDARD=17", *std_cmake_args
       system "cmake", "--build", ".", "--target", "install"
     end
```

Comment:
```suggestion
    system "cmake", "-S", ".", "-B", "build", "-DCMAKE_CXX_STANDARD=17", *std_cmake_args
    system "cmake", "--install", "build"
```
We can remove `BUILD_TESTING` since that is now part of `std_cmake_args`.